docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,893 @@
+  concise without losing meaning. Although users are more likely to set this parameter using an environment variable.
+- In the `wsv` section, the field `wasm_runtime_config` has a redundant `_config` suffix. Given these are all
+  configuration parameters, such suffixes can be eliminated for clarity.
+
+Implementing a consistent naming convention like this will simplify the user's configuration process and reduce
+ambiguity.
+
+#### Redundant SCREAMING_CASE in JSON Configuration
+
+The adoption of SCREAMING_CASE for parameters within the JSON configuration appears out of place. Such uppercase field
+names might be customary for environment variables, but in a JSON setting, they don't offer any distinctive benefit.
+Here's a snippet from the current configuration showcasing this:
+
+```json
+{
+  "PUBLIC_KEY": null,
+  "PRIVATE_KEY": null,
+  "DISABLE_PANIC_TERMINAL_COLORS": false
+}
+```
+
+---
+
+In summary, these issues underscore the need for clearer, more intuitive naming conventions in Iroha 2's configuration.
+
+### Rustisms in the Configuration Reference
+
+Configuration reference contains Rust-specific expressions that wouldn't make sense to the end-user who is configuring
+Iroha and might not be familiar with Rust. Here are some examples:
+
+- [Using `Option<..>`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#option)
+- [Using `std::path::PathBuf`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#loggerlog_file_path)
+
+The use of rustisms like that overcomplicates the configuration reference and affects the user experience. Further, it
+doesn't actually provide any extra information to a user familiar with Rust, as environment variables and JSON values
+cannot be `None` due to being stringly typed and null-able regardless of whether the type specifies `Option`.
+
+### Unhelpful Error Messages
+
+Some of the error messages related to Iroha configuration are inconsistent and misleading. Here's a quick rundown of
+various scenarios, associated errors and possible confusions.
+
+#### 1. No configuration file
+
+Running Iroha without a configuration file results in the following log and error:
+
+```
+IROHA_TORII: environment variable not found
+IROHA_SUMERAGI: environment variable not found
+IROHA_KURA: environment variable not found
+IROHA_BLOCK_SYNC: environment variable not found
+IROHA_PUBLIC_KEY: environment variable not found
+IROHA_PRIVATE_KEY: environment variable not found
+IROHA_GENESIS: environment variable not found
+Configuration file not found. Using environment variables as fallback.
+Error:
+   0: Please add `PUBLIC_KEY and PRIVATE_KEY` to the configuration.
+```
+
+Several things are wrong with this error message:
+
+- The information about using environment variables as fallback comes after the messages about them not being found.
```

Comment:
moreover, there shouldn't be any explicit mention of an environment variable in an error, a config file is recommended approach, an env variable is just an alternative, hence generic 'value not set' should be mentioned

Reply:
I agree with @0x009922 here. If we don't see which field has an issue, how do we debug an Iroha deployment?